### PR TITLE
Fixes issue-11740: Added support for the om service to connect to AWS RDS using IAM roles

### PR DESCRIPTION
**Pull request by ArtyomyuS**
### Describe your changes:

Fixes [issue-11740](https://github.com/open-metadata/OpenMetadata/issues/11740)

I worked on adding support for the OM service to be able to connect to AWS RDS PostgreSQL using IAM roles authentication because the exposure of the USERNAME and PASSWORD on the Env variables is a security leak. 

Changes made:

- Added a new spec for the AWS configurations (region, enableIamDatabaseAuthentication).
- Added maven dependencies related to aws authentication
- Updated TablesInitializer to check based on jdbc URL if additional authentication steps are required.
- Added `DatabaseAuthenticationProvider` interface to be able to implement different auth providers. Currently only AWS RDS supported.
- Removed `DB_USE_SSL` from the sql paramas and added a more generic parameter `DB_PARAMS` to be able to support for both mysql and postgresql required parameters by each db. Default remain as it was.

No new tests being added as the logic rely on the application initialisation for the Jdbi that shall support same database configuration properties. 

#
### Type of change:
- [ ] Bug fix
- [ ] Improvement
- [x] New feature
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [x] Documentation

#
### Checklist:
- [x] I have read the [**CONTRIBUTING**](https://docs.open-metadata.org/developers/contribute) document.
- [x] My PR title is `Fixes [issue-11740](https://github.com/open-metadata/OpenMetadata/issues/11740)`
- [x] I have commented on my code, particularly in hard-to-understand areas. 
- [ ] For JSON Schema changes: I updated the migration scripts or explained why it is not needed.

New feature
- [x] The issue properly describes why the new feature is needed, what's the goal, and how we are building it. Any discussion
    or decision-making process is reflected in the issue.
- [x] I have updated the documentation.
- [ ] I have added tests around the new logic.

### CODE REVIEW

**Comment by pmbrull** [comment]
hi @ArtyomyuS thanks for the PR! I'll let @harshach / @mohityadav766 pitch in on the code!

From my end just a couple of questions:
1. Following up on our discussion on the issue, should we add [sqlalchemy-rdsiam](https://github.com/cisco-open/sqlalchemy-rdsiam/tree/main) dependency on the ingestion images?
2. Maybe related to the point above, if you could please prep some [docs](https://github.com/open-metadata/OpenMetadata/tree/main/openmetadata-docs/content/v1.1.0-snapshot) on how to set this up, that would be great since you have already validated the setup during the development.

Many thanks!!

**Comment by ArtyomyuS** [comment]
> hi @ArtyomyuS thanks for the PR! I'll let @harshach / @mohityadav766 pitch in on the code!
> 
> From my end just a couple of questions:
> 
> 1. Following up on our discussion on the issue, should we add [sqlalchemy-rdsiam](https://github.com/cisco-open/sqlalchemy-rdsiam/tree/main) dependency on the ingestion images?
> 2. Maybe related to the point above, if you could please prep some [docs](https://github.com/open-metadata/OpenMetadata/tree/main/openmetadata-docs/content/v1.1.0-snapshot) on how to set this up, that would be great since you have already validated the setup during the development.
> 
> Many thanks!!

Hello @pmbrull , thanks for your reply. 

1. I'll try to add also a how to section in the docs. 
2. The dependencies for the [sqlalchemy-rdsiam](https://github.com/cisco-open/sqlalchemy-rdsiam/tree/main) we believe may not be required as we want to consider aws airflow integration here that is already supported by the OM.
3. Going to fix javadocs

File: docker/openmetadata.yaml
Comment (harshach) [inline docker/openmetadata.yaml:128]: @ArtyomyuS Thank for the PR. Any reason to remove DB_USE_SSL and making useSSL=false.
This will prevent others from enabling useSSL option.

File: openmetadata-service/src/main/java/org/openmetadata/service/OpenMetadataApplicationConfig.java
Comment (harshach) [inline openmetadata-service/src/main/java/org/openmetadata/service/OpenMetadataApplicationConfig.java:50]: @ArtyomyuS this should be optional and nullable. 
i.e users who don't pass this section should continue to run

File: openmetadata-service/src/main/java/org/openmetadata/service/util/AWSUtils.java
Comment (harshach) [inline openmetadata-service/src/main/java/org/openmetadata/service/util/AWSUtils.java:41]: do we need this check, LOG.debug should do this by default

File: openmetadata-service/src/main/java/org/openmetadata/service/util/JdbiUtils.java
Comment (harshach) [inline openmetadata-service/src/main/java/org/openmetadata/service/util/JdbiUtils.java:39]: can we call this name as checkJdbiAuthenticationProvider, as we use AuthenticationProvider elsewhere for different use-cases

File: openmetadata-service/src/main/java/org/openmetadata/service/util/JdbiUtils.java
Comment (harshach) [inline openmetadata-service/src/main/java/org/openmetadata/service/util/JdbiUtils.java:44]: what's the substring doing here and why there is 5 chars. Can we add some comment on this number

**Comment by harshach** [comment]
Thanks @ArtyomyuS for the PR. Overall looks great!! left some minor nits. After you address those we can merge it in

File: docker/openmetadata.yaml
Comment (ArtyomyuS) [reply docker/openmetadata.yaml:128]: Hello @harshach , thanks for your comment. The idea was to extract all the params out of the db url specification as MySQL and PostgreSQL have different params. In case of `useSSL` this is supported just by MySQL, in case of PostgreSQL it is: `sslmode`. So I thought that is more appropriate just to specify whatever parameters are required based on the desired db integration.

File: openmetadata-service/src/main/java/org/openmetadata/service/OpenMetadataApplicationConfig.java
Comment (ArtyomyuS) [reply openmetadata-service/src/main/java/org/openmetadata/service/OpenMetadataApplicationConfig.java:50]: Ok, going to add it as optional

File: openmetadata-service/src/main/java/org/openmetadata/service/util/AWSUtils.java
Comment (ArtyomyuS) [reply openmetadata-service/src/main/java/org/openmetadata/service/util/AWSUtils.java:41]: Going to change that

File: openmetadata-service/src/main/java/org/openmetadata/service/util/JdbiUtils.java
Comment (ArtyomyuS) [reply openmetadata-service/src/main/java/org/openmetadata/service/util/JdbiUtils.java:39]: Will do

File: openmetadata-service/src/main/java/org/openmetadata/service/OpenMetadataApplicationConfig.java
Comment (ArtyomyuS) [reply openmetadata-service/src/main/java/org/openmetadata/service/OpenMetadataApplicationConfig.java:50]: @harshach isn't it by default optional btw, by specifying `@JsonProperty(value = "aws", required = false)` it looks to be redundant as the default is always `false` for the `required` parameter.

File: docker/openmetadata.yaml
Comment (harshach) [reply docker/openmetadata.yaml:128]: @ArtyomyuS that makes sense. can we remove the useSSL=false from the connection string? otherwise it will override what we are passing through DB_PARAMS right?

File: docker/openmetadata.yaml
Comment (harshach) [inline docker/openmetadata.yaml:130]: I think it would be better if we move this section under database config, otherwise users might think this unrelated config for db @ArtyomyuS

**Comment by harshach** [comment]
> Hello @harshach, any thoughts on the updated content?

@ArtyomyuS sorry! didn't notice your comments earlier. Left a follow-up can you take a look and address. Thanks

File: docker/openmetadata.yaml
Comment (ArtyomyuS) [reply docker/openmetadata.yaml:130]: Hello @harshach , thank you. Going to change that 👍

**Comment by ArtyomyuS** [comment]
Hello @harshach , did the changes, please let me know.

**Comment by ArtyomyuS** [comment]
Hello @harshach , kind reminder :)

**Review by harshach — APPROVED** [review]

**Comment by harshach** [comment]
@ArtyomyuS looks good to me. Can you check why the tests are failing

**Comment by ArtyomyuS** [comment]
Hello @pmbrull and @harshach ,
 sorry for late update on this merge-request. I noticed that those test are failing due to the following step that does not bootstrap ES properly and remain constantly in Checking phase. Would you please advice or recommend a fix here. Thanks.
```
until curl -s -f "http://localhost:9200/_cat/indices/team_search_index"; do
  printf 'Checking if Elastic Search instance is up...\n'
  sleep 5
done
```

**Comment by ArtyomyuS** [comment]
Hello. Did another merge from main branch. @harshach any thoughts on the above comment? thanks

**Comment by ArtyomyuS** [comment]
Going to change the bootstrap of the DatabaseFactory to consider RDS authentication based on the provided URL as the `database.aws` cannot be deserialised by the dropwizard class. The actual pull-request is a DRAFT at this moment. Going to update it accordingly to new findings.

**Comment by cristiancalugaru** [comment]
@ArtyomyuS thanks for starting this work. I have a similar requirement - but I'm working with RDS backed by Mysql not Postgresql. I see you mention Postgresql in the description - but the code looks pretty much agnostic when it comes to what type of RDS is actually used. Would this work for both?

Let me know if I can help out with something to push this forward. Thanks!

**Comment by ArtyomyuS** [comment]
> @ArtyomyuS thanks for starting this work. I have a similar requirement - but I'm working with RDS backed by Mysql not Postgresql. I see you mention Postgresql in the description - but the code looks pretty much agnostic when it comes to what type of RDS is actually used. Would this work for both?
> 
> Let me know if I can help out with something to push this forward. Thanks!

Hello, yes it shall work for any RDS

**Comment by ArtyomyuS** [comment]
The pull-requst being udpated accordingly. Waiting for another round of review from @harshach. Thank you

**Comment by ArtyomyuS** [comment]
@cristiancalugaru  you can try building an image from this branch to test the functionality. Please let me know if you encounter any issues related to Mysql RDS config. You could check out the configuration related to devops building procedure.

File: openmetadata-docs/content/v0.13.3/how-to-guides/aws/index.md
Comment (ArtyomyuS) [reply openmetadata-docs/content/v0.13.3/how-to-guides/aws/index.md:4]: Ok, going to update it. Thanks

File: conf/openmetadata.yaml
Comment (allenkallz) [inline conf/openmetadata.yaml:128]: Hi 
May i know the reason why useSSL is set to False and not reading from the DB_USE_SSL env variable.

Thanks

**Pull request merged**
